gui/gui/src/ts/components/search/NodeLabelSearch.tsx

```diff
@@ -0,0 +1,37 @@
+import * as React from "react";
+
+import {Paper, InputAdornment, InputBase} from "@material-ui/core";
+
+import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
+import {faSearch} from "@fortawesome/free-solid-svg-icons";
+
+export const NodeLabelSearch: React.FunctionComponent<{
+  onSubmit: (value: string) => void;
+}> = ({onSubmit}) => {
+  const searchInputRef = React.useRef<HTMLInputElement>(null);
```

Comment:
Use a controlled input and state rather than a ref. Refs should be avoided whenever possible.
See
https://github.com/minorg/paradicms/blob/master/app/generic/gui/src/ts/paradicms/app/generic/components/navbar/NavbarSearchForm.tsx

Reply:
See this for an overview of controlled components:
https://reactjs.org/docs/forms.html

Reply:
Gotcha changed